analyzers/rspec/cs/S1168_c#.html

```diff
@@ -1,16 +1,19 @@
-<p>Returning <code>null</code> instead of an actual array, collection or map forces callers of the method to explicitly test for nullity, making them
-more complex and less readable.</p>
-<p>Moreover, in many cases, <code>null</code> is used as a synonym for empty.</p>
+<p>Returning <code>null</code> or <code>default</code> instead of an actual collection forces the method callers to explicitly test for null, making
+the code more complex and less readable.</p>
+<p>Moreover, in many cases, <code>null</code> or <code>default</code> is used as a synonym for empty.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
 public Result[] GetResults()
 {
     return null; // Noncompliant
 }
 
-public IEnumerable&lt;Result&gt; GetResults()
+public IEnumerable&lt;Result&gt; GetResults(bool condition)
 {
-    return null; // Noncompliant
+    var results = GenerateResults();
```

Comment:
review note: the functional changes have been done in PR https://github.com/SonarSource/sonar-dotnet/pull/6457